### PR TITLE
WIP - Adding watch files using gulp

### PR DESCRIPTION
**Pull request by rajzshkr**
This is a initial draft, Please add your comments to make it better.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+const gulp = require('gulp');
+const watch = require('gulp-watch');
+const babel = require('gulp-babel');
+
+const babelrc = JSON.parse(fs.readFileSync('./.babelrc', 'utf8'));
+
+// Make it as a util, since used in multiple place
+var ver = process.versions.node;
+var majorVer = parseInt(ver.split('.')[0], 10);
+const buildType = majorVer < 8 ? 'legacy' : 'modern';
```
Comment (rajzshkr) [inline gulpfile.js]: <= ?

Comment (ajaymathur) [reply gulpfile.js]: < is good

**Comment by thejameskyle** [comment]
Yuck why?

Also no need to add gulp, Babel has a watch mode

**Comment by rajzshkr** [comment]
@thejameskyle I thought of making the contributor life easy, instead of building the dist every time.

**Pull request closed without merging**